packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1211,7 +1211,11 @@ abstract class ScribbleClient {
 class SelectionRect {
   /// Constructor for creating a [SelectionRect] from a text [position] and
   /// [bounds].
-  const SelectionRect({required this.position, required this.bounds});
+  const SelectionRect({
+    required this.position,
+    required this.bounds,
+    this.direction = TextDirection.ltr,
```

Comment:
maybe better to remove the default value here in case we incorrectly assume the writing direction?